From c7cc2483d591784bb2707f72b0784d92074f5347 Mon Sep 17 00:00:00 2001
From: Keith Packard <keithp@keithp.com>
Date: Tue, 29 Apr 2025 21:31:38 -0700
Subject: [PATCH] scripts: Support qemu version 10

Adapt command line option code to include qemu version 10 too.

Signed-off-by: Keith Packard <keithp@keithp.com>
---
 scripts/run-riscv | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/scripts/run-riscv b/scripts/run-riscv
index 40f3863e07..076aea4772 100755
--- a/scripts/run-riscv
+++ b/scripts/run-riscv
@@ -84,11 +84,11 @@ fi
 
 all_options="i e g m a f d c s u"
 
-if $qemu --version | grep -q 'version [789]'; then
+if $qemu --version | grep -q 'version \(7\|8\|9\|10\)'; then
     all_options="h $all_options"
 fi
 
-if $qemu --version | grep -q 'version [89]'; then
+if $qemu --version | grep -q 'version \(8\|9\|10\)'; then
     if $qemu --version | grep -q 'version 8.1'; then
         all_options="Zawrs $all_options"
     else
@@ -96,7 +96,7 @@ if $qemu --version | grep -q 'version [89]'; then
     fi
 fi
 
-if $qemu --version | grep -q 'version \(8.[1-9]\|9\)'; then
+if $qemu --version | grep -q 'version \(8.[1-9]\|9\|10\)'; then
     all_options="zfa $all_options"
 fi
 
-- 
2.49.0

